api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_online.py

```diff
@@ -74,12 +74,20 @@ def __init__(self, n_samples: int, n_variables: int):
 
         self.u = np.zeros(n_variables, dtype=np.float64)
         self.n_samples = n_samples
+        # If we want to exclude zeros, we need to keep track of the denominator
+        self.exclude_zeros = exclude_zeros
+        self.denominator = np.zeros(n_variables)
 
     def update(self, var_vec: npt.NDArray[np.int64], val_vec: npt.NDArray[np.float32]) -> None:
```

Comment:
This approach requires two passes through the `var_vec` if `exclude_zeros is True`.  I am almost certain a faster approach would be to have two separate `_update_...` functions, one which updates only the `u` accumulator, and the other which updates _both_ `u` and `demoninator`.

In other words, this `update` method would look like:
```
def update(self, var_vec: npt.NDArray[np.int64], val_vec: npt.NDArray[np.float32]) -> None:
    if self.exclude_zeros:
        _update_mean_and_denominator_vectors(self.u, self.denominator, var_vec, val_vec)
    else:
        _update_mean_vector(self.u, var_vec, val_vec)
```

The implementation of `_update_mean_vector` is unchanged.  The `_update_mean_and_denominator_vectors` performs _both_ operations within a single loop, e.g.,
```
def _update_mean_and_denominator_vector(
    u: npt.NDArray[np.float64],
    denominator: npt.NDArray[np.float64],
    var_vec: npt.NDArray[np.int64],
    val_vec: npt.NDArray[np.float32],
) -> None:
    for col, val in zip(var_vec, val_vec):
        u[col] = u[col] + val
        denominator[col] += 1
```